File: sidekiq/enqueue_jobs.rb
require "redis"
require "json"
require "securerandom"

TOTAL_JOBS = Integer(ENV.fetch("TOTAL_JOBS", 200_000))
NUM_QUEUES = Integer(ENV.fetch("NUM_QUEUES", 10))

redis = Redis.new(url: "redis://localhost:6379")

start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)

TOTAL_JOBS.times do |i|
  queue = "queue_#{i % NUM_QUEUES}"
  job = {
    "class" => "LoadTestJob",
    "args" => [i],
    "queue" => queue,
    "jid" => SecureRandom.hex(12),
    "created_at" => Time.now.to_f,
    "enqueued_at" => Time.now.to_f,
    "retry" => false,
  }.to_json

  redis.lpush("queue:#{queue}", job)
  redis.sadd("queues", queue)
end

end_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
puts "All jobs enqueued within: #{end_time - start_time} seconds"

# Memory stats
info = redis.info("memory")
puts "=== MEMORY AT PEAK (before consuming) ==="
puts "  used_memory_human:      #{info['used_memory_human']}"
puts "  used_memory_peak_human: #{info['used_memory_peak_human']}"
puts "  used_memory_rss_human:  #{info['used_memory_rss_human']}"
puts "=========================================="

File: laravel/enqueue_jobs.php
<?php

require __DIR__ . "/vendor/autoload.php";

$app = require_once __DIR__ . "/bootstrap/app.php";
$app->make(Illuminate\Contracts\Console\Kernel::class)->bootstrap();

$totalJobs = (int) (getenv("TOTAL_JOBS") ?: 200000);
$numQueues = (int) (getenv("NUM_QUEUES") ?: 10);

$startTime = microtime(true);

for ($i = 0; $i < $totalJobs; $i++) {
    $queue = "queue_" . ($i % $numQueues);
    App\Jobs\LoadTestJob::dispatch($i)->onQueue($queue);
}

$endTime = microtime(true);
printf("All jobs enqueued within: %.2f seconds\n", $endTime - $startTime);

$redis = app("redis")->connection()->client();
$info = $redis->info("memory");
echo "=== MEMORY AT PEAK (before consuming) ===\n";
echo "  used_memory_human:      " . $info["used_memory_human"] . "\n";
echo "  used_memory_peak_human: " . $info["used_memory_peak_human"] . "\n";
echo "  used_memory_rss_human:  " . $info["used_memory_rss_human"] . "\n";
echo "==========================================\n";
